### PR TITLE
Add a test for FLIP_Y and PREMULTIPLY_ALPHA behaviors in WebGL2.

### PR DESCRIPTION
**Pull request by zhenyao**
Also update the 2.0 spec.

### CODE REVIEW

**Comment by zhenyao** [comment]
@kenrussell @kainino0x @jdashg PTAL
@grorg @RafaelCintron FYI (and also feel free to review)

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+
+function getPixelsFromTexture2D(gl, tex, xoffset, yoffset, width, height) {
+  var fbo = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+  gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, tex, 0);
+  var bytesPerReadbackPixel = 4;
+  var readbackBuffer = setupArrayBuffer(width * height * bytesPerReadbackPixel, false);
+  gl.readPixels(xoffset, yoffset, width, height, gl.RGBA, gl.UNSIGNED_BYTE, readbackBuffer);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "read back texture pixels should succeed");
+  var bytesPerPixel = 4;
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Is there a strong reason for making a second copy of this data? Is the intent to bypass any pixel pack parameters, or similar? If this is just copied code from a different test then let's simplify it here.

Comment (zhenyao) [reply sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: You are right. I copied from another test where the readbackBuffer is RGBA but the data buffer is RGB. We don't need to make another copy here. Revised.

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+    return;
+  }
+  var count = 0;
+  for (var ii = 0; ii < buffer1.length / 4; ++ii) {
+    var distance = 0;
+    for (var jj = 0; jj < 4; ++jj) {
+      var diff = buffer1[ii * 4 + jj] - buffer2[ii * 4 + jj];
+      distance += diff * diff;
+    }
+    if (Math.sqrt(diff) > tolerance) {
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Ooh, math! :D This seems fine; other tests have used absolute differences, but this is probably better.

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "texImage3D from pbo with PREMULTIPLY_ALPHA=true should fail");
+  gl.texSubImage3D(gl.TEXTURE_3D, 0, 0, 0, 0, width, height, depth, gl.RGBA, gl.UNSIGNED_BYTE, array);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "texSubImage3D from pbo with PREMULTIPLY_ALPHA=true should fail");
+  gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, false);
+}
+
+function runRegressionTests() {
+  debug("");
+  debug("The following test serves as a regression test for crbug.com/774174");
+  var ele = document.createElementNS('http://www.w3.org/1999/xhtml', 'canvas');
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Is this really necessary? Can we just use:
  document.createElement('canvas')
?

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+  var context = ele.getContext('webgl2', {alpha: false, depth: false, stencil: true, premultipliedAlpha: true});
```
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Can we reuse the context from the other tests and just reset all of the pixel unpack parameters? If that makes the test too hard to understand, never mind my question.

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html (hunk 3 above)
Comment (zhenyao) [reply sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Addressed.

File: sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html (hunk 4 above)
Comment (zhenyao) [reply sdk/tests/conformance2/textures/misc/tex-unpack-params-with-flip-y-and-premultiply-alpha.html]: Addressed.

**Comment by zhenyao** [comment]
All comments addressed. Merging myself.

**Pull request merged**